### ISSUE TITLE
Test failures with libyaml 0.2.3

### ISSUE THREAD

**Issue by jeckersb** [yaml/pyyaml#396]
```
===========================================================================
test_emitter_styles_ext(tests/data/spec-07-09.data, tests/data/spec-07-09.canonical): FAILURE
Traceback (most recent call last):
  File "tests/lib3/test_appliance.py", line 59, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib3/test_yaml_ext.py", line 256, in wrapper
    function(*args, **kwds)
  File "tests/lib3/test_emitter.py", line 62, in test_emitter_styles
    _compare_events(events, new_events)
  File "tests/lib3/test_emitter.py", line 15, in _compare_events
    assert event1.value == event2.value, (event1, event2)
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='foo'),
 ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='foo %YAML 1.1'))
---------------------------------------------------------------------------
tests/data/spec-07-09.data:
---
foo
...
# Repeated end marker.
...
---
bar
# No end marker.
---
baz
...
---------------------------------------------------------------------------
tests/data/spec-07-09.canonical:
%YAML 1.1
---
!!str "foo"
%YAML 1.1
---
!!str "bar"
%YAML 1.1
---
!!str "baz"
===========================================================================
test_emitter_styles_ext(tests/data/spec-07-10.data, tests/data/spec-07-10.canonical): FAILURE
Traceback (most recent call last):
  File "tests/lib3/test_appliance.py", line 59, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib3/test_yaml_ext.py", line 256, in wrapper
    function(*args, **kwds)
  File "tests/lib3/test_emitter.py", line 62, in test_emitter_styles
    _compare_events(events, new_events)
  File "tests/lib3/test_emitter.py", line 15, in _compare_events
    assert event1.value == event2.value, (event1, event2)
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='Root flow scalar'),
 ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='Root flow scalar %YAML 1.1'))
---------------------------------------------------------------------------
tests/data/spec-07-10.data:
"Root flow
 scalar"
--- !!str >
 Root block
 scalar
---
# Root collection:
foo : bar
... # Is optional.
---
# Explicit document may be empty.
---------------------------------------------------------------------------
tests/data/spec-07-10.canonical:
%YAML 1.1
---
!!str "Root flow scalar"
%YAML 1.1
---
!!str "Root block scalar\n"
%YAML 1.1
---
!!map {
  ? !!str "foo"
  : !!str "bar"
}
---
#!!str ""
!!null ""
===========================================================================
test_emitter_styles_ext(tests/data/spec-07-13.data, tests/data/spec-07-13.canonical): FAILURE
Traceback (most recent call last):
  File "tests/lib3/test_appliance.py", line 59, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib3/test_yaml_ext.py", line 256, in wrapper
    function(*args, **kwds)
  File "tests/lib3/test_emitter.py", line 62, in test_emitter_styles
    _compare_events(events, new_events)
  File "tests/lib3/test_emitter.py", line 15, in _compare_events
    assert event1.value == event2.value, (event1, event2)
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(ScalarEvent(anchor=None, tag='!foo', implicit=(False, False), value='No directives'),
 ScalarEvent(anchor=None, tag='!foo', implicit=(False, False), value='No directives %TAG ! %21foo'))
---------------------------------------------------------------------------
tests/data/spec-07-13.data:
! "First document"
---
!foo "No directives"
%TAG ! !foo
---
!bar "With directives"
%YAML 1.1
---
!baz "Reset settings"
---------------------------------------------------------------------------
tests/data/spec-07-13.canonical:
%YAML 1.1
---
!!str "First document"
---
!<!foo> "No directives"
---
!<!foobar> "With directives"
---
!<!baz> "Reset settings"
===========================================================================
test_emitter_styles_ext(tests/data/spec-08-08.data, tests/data/spec-08-08.canonical): FAILURE
Traceback (most recent call last):
  File "tests/lib3/test_appliance.py", line 59, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib3/test_yaml_ext.py", line 256, in wrapper
    function(*args, **kwds)
  File "tests/lib3/test_emitter.py", line 62, in test_emitter_styles
    _compare_events(events, new_events)
  File "tests/lib3/test_emitter.py", line 15, in _compare_events
    assert event1.value == event2.value, (event1, event2)
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='foo bar'),
 ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='foo bar %YAML 1.1'))
---------------------------------------------------------------------------
tests/data/spec-08-08.data:
---
foo:
 "bar
 baz"
---
"foo
 bar"
---
foo
 bar
--- |
 foo
...
---------------------------------------------------------------------------
tests/data/spec-08-08.canonical:
%YAML 1.1
---
!!map {
  ? !!str "foo"
  : !!str "bar baz"
}
%YAML 1.1
---
!!str "foo bar"
%YAML 1.1
---
!!str "foo bar"
%YAML 1.1
---
!!str "foo\n"
===========================================================================
TESTS: 2614
FAILURES: 4
```

**Comment by perlpunk** [comment]
Thanks! Seems I did not get the open_ended logic right...

**Comment by perlpunk** [comment]
There's a fix: https://github.com/yaml/libyaml/pull/174
libyaml release 0.2.4 is being prepared.

**Comment by perlpunk** [comment]
Hopefully this won't happen again. I added a `pyyaml-test` target to the `packaging` branch, that can be run before the release.

**Comment by perlpunk** [comment]
Released libyaml 0.2.4!
